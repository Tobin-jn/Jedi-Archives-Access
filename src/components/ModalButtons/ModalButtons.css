.modal-buttons-wrapper {
  position: absolute;
  background: rgba(0, 0, 0, .9);
  z-index: 3;
  width: 100vw;
  height: 100vh;
  padding-left: 1rem;
  display: none;
  top: 100px;
  left: -500px;
  transition: .5s;
}

.modal-wrapper .deployed {
  left: 0;
}

.modal-buttons-wrapper .nav-button {
  text-align: default;
}

@media (max-width: 475px){
  .modal-buttons-wrapper {
    display: block;
  }

  .nav-button-text {
    text-align: start;
    font-size: 1rem;
    width: 125px;
  }

  .nav-button {
    width: 125px;
  }

  .nav-button-text:before {
    width: 125px;
    font-size: .8rem;
    left: 0;
  }
}