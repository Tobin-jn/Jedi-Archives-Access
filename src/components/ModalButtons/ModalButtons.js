import React from 'react';
import Button from '../Button/Button';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import './ModalButtons.css';

const ModalButtons = ({ currentSelection, handleSelection, hamburger }) => {

  return(
    <aside 
      className={`modal-buttons-wrapper ${hamburger.status}`}
      aria-label='mobile-navigation-section'
    >
      <NavLink 
        to='/people' 
        className='nav-button people' 
      >
        <Button 
          currentSelection={currentSelection}
          buttonName='people'
          handleSelection={handleSelection}
        />
      </NavLink>
      <NavLink 
        to='/planets' 
        className='nav-button planets' 
      >
        <Button 
          currentSelection={currentSelection}
          buttonName='planets'
          handleSelection={handleSelection}
        />
      </NavLink>
      <NavLink 
        to='/vehicles' 
        className='nav-button vehicles' 
      >
        <Button 
          currentSelection={currentSelection}
          buttonName='vehicles'
          handleSelection={handleSelection}
        />
      </NavLink>
      <NavLink 
        to='/favorites' 
        className='nav-button favorites' 
      >
        <Button 
          currentSelection={currentSelection}
          buttonName='saved'
          handleSelection={handleSelection}
        />
      </NavLink>
    </aside>
  );
}

ModalButtons.propTypes = {
  currentSelection: PropTypes.string.isRequired,
  handleSelection: PropTypes.func.isRequired,
  hamburger: PropTypes.object.isRequired,
};

export default ModalButtons;