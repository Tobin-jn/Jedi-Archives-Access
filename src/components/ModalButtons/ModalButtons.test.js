import React from 'react';
import ModalButtons from './ModalButtons';
import { shallow } from 'enzyme';

let wrapper;
let mockCurrentSelection = 'people'
let mockHandleSelection = jest.fn();
let mockHamburger = ''

beforeEach(() => {
  wrapper = shallow(<ModalButtons 
    currentSelection={mockCurrentSelection}
    handleSelection={mockHandleSelection}
    hamburger={mockHamburger}
  />)
}) 

describe('ModalButtons', () => {
  it ('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});