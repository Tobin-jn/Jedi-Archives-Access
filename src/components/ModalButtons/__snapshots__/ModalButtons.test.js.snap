// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalButtons should match the snapshot 1`] = `
<aside
  className="modal-buttons-wrapper undefined"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav-button people"
    to="/people"
  >
    <Button
      buttonName="people"
      currentSelection="people"
      handleSelection={[MockFunction]}
    />
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav-button planets"
    to="/planets"
  >
    <Button
      buttonName="planets"
      currentSelection="people"
      handleSelection={[MockFunction]}
    />
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav-button vehicles"
    to="/vehicles"
  >
    <Button
      buttonName="vehicles"
      currentSelection="people"
      handleSelection={[MockFunction]}
    />
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav-button favorites"
    to="/favorites"
  >
    <Button
      buttonName="saved"
      currentSelection="people"
      handleSelection={[MockFunction]}
    />
  </NavLink>
</aside>
`;
